physionet-django/events/views.py

```diff
@@ -54,21 +59,45 @@ def event_home(request):
     url_prefix = notification.get_url_prefix(request)
 
     form_error = False
-    if request.method == 'POST':
+    if request.method == 'POST' and 'add-event' in request.POST.keys():
         event_form = AddEventForm(user=user, data=request.POST)
         if event_form.is_valid() and is_instructor:
             event_form.save()
             return redirect(event_home)
         else:
             form_error = True
 
+    # handle notifications to join an event
+    if request.method == 'POST' and 'participation_response' in request.POST.keys():
+
+        formset = EventApplicationResponseFormSet(request.POST)
+        # only process the form that was submitted
+        for form in formset:
+            if form.instance.id == int(request.POST['participation_response']):
+                if form.is_valid():
+                    event_application = form.save(commit=False)
+                    event_application.accept(comment=form.cleaned_data.get('decision_comment'))
```

Comment:
Is this the correct place to be calling "accept"? It looks like people are being added to the list of participants even when applications are rejected.